src/Doctrine/EventListener/PurgeHttpCacheListener.php

```diff
@@ -127,7 +126,7 @@ private function gatherResourceAndItemTags(object $entity, bool $purgeItem): voi
 
     private function gatherRelationTags(EntityManagerInterface $em, object $entity): void
     {
-        $associationMappings = $em->getClassMetadata(ClassUtils::getClass($entity))->getAssociationMappings();
+        $associationMappings = $em->getClassMetadata($em->getClassMetadata(get_class($entity))->getName())->getAssociationMappings();
```

Comment:
why the double call to getClassMetadata?

Reply:
Call it a long day lol :-), I'll get around to fixing this pr up this week properly with tests. Ran it on latest doctrine orm with symfony 7 just to see how far we are and this caught my attention, tests failed on 3.2 to begin with, with these dependencies so I'd rather wait to rebase and sort it out.